### PR TITLE
include searchtips modal

### LINKED ISSUE #2487: Help text needed to explain to users what advanced search features are available
Expected behavior

Provide help text to let users know what advanced search features are available to them, with an explanation of Boolean searching which is enabled. For implementation, use whatever help text feature is a11y compliant.

### PR DESCRIPTION
**Pull request by hudajkhan**
Closes #2487 

What this PR does:
- Adds a component that displays the link that will open up the search tips modal
- Adds a component that displays the content within the modal
- Include a path for search tips link to open a modal with search tips content
- Overrides exhibit navbar in order to also display the search tips link next to the search bar
- Include tests for the component displaying the link for the modal and for the component responsible for the modal content.
- Adds styling overrides 
- Also add ViewComponent configuration in spec_helper

Also, some notes:
I asked @alundgard and it is ok if the close button has a red outline for now.  Also, the mobile view shows the search tips button on top of the search bar and not the bottom (which is what the designs show).  This is also ok for now.

### CODE REVIEW

File: app/assets/stylesheets/modules/spotlight_overrides.scss
Comment (corylown) [inline app/assets/stylesheets/modules/spotlight_overrides.scss:320]: Are these styles needed? The link text looks the same if I remove them and the icon position seems okay.

**Review by corylown — CHANGES_REQUESTED** [review]

File: app/components/exhibits/search_tips_link_component.html.erb
Comment (corylown) [inline app/components/exhibits/search_tips_link_component.html.erb:7]: There's an info icon component already in spotlight that could be used via `blacklight_icon('info')`. Should we use that icon or override it locally with this one?

File: spec/spec_helper.rb
Comment (hudajkhan) [reply spec/spec_helper.rb:38]: I will make these changes

File: app/assets/stylesheets/modules/spotlight_overrides.scss
Comment (hudajkhan) [reply app/assets/stylesheets/modules/spotlight_overrides.scss:320]: The first style applies when we add an image to the masthead.  When that happens, there is an existing style that changes the text to uppercase.  The style on line 314 prevents that from happening for search tips, since these should stay lowercase even with an image masthead.

For the second style, the vertical spacing did not seem exactly where the Figma screen had it.

File: app/components/exhibits/search_tips_link_component.html.erb
Comment (hudajkhan) [reply app/components/exhibits/search_tips_link_component.html.erb:7]: Happy to use an existing one. I'll try that out, thanks.

File: app/assets/stylesheets/modules/spotlight_overrides.scss
Comment (corylown) [reply app/assets/stylesheets/modules/spotlight_overrides.scss:320]: Okay. It's really strange that spotlight upcases the nav links when there's background image, but otherwise doesn't.

File: spec/spec_helper.rb
Comment (hudajkhan) [reply spec/spec_helper.rb]: I have made these changes

File: app/components/exhibits/search_tips_link_component.html.erb
Comment (hudajkhan) [reply app/components/exhibits/search_tips_link_component.html.erb]: I set one up using the blacklight icon method. This also changed some styling (there's an extra space that the component adds within the blacklight icon next to the svg element)

**Comment by hudajkhan** [comment]
Requires waiting for approval of https://github.com/projectblacklight/spotlight/pull/3171 with some additional changes in exhibits

**Comment by hudajkhan** [comment]
Screenshots of current view in local development:

Screenshot 2024-10-08 at 2 36 59 PM

Screenshot 2024-10-08 at 2 37 08 PM

File: app/assets/stylesheets/modules/spotlight_overrides.scss
Comment (corylown) [inline app/assets/stylesheets/modules/spotlight_overrides.scss:323]: Is this doing anything? I don't see any underlined whitespace when I remove this?

Comment (hudajkhan) [reply app/assets/stylesheets/modules/spotlight_overrides.scss]: It was displaying an underline earlier but now no longer is (perhaps due to upstream changes in Blacklight?)  I've removed this line since it no longer appears necessary.

**Review by corylown — APPROVED** [review]

**Pull request merged**
